astronomer/providers/google/cloud/hooks/dataproc.py

```diff
@@ -29,9 +29,18 @@ def get_cluster_client(
         :param location: (To be deprecated). The Cloud Dataproc region in which to handle the request.
         """
         client_options, region = self._get_client_options_and_region(region=region, location=location)
-        return ClusterControllerAsyncClient(
-            credentials=self._get_credentials(), client_info=CLIENT_INFO, client_options=client_options
-        )
+        try:
+            # for apache-airflow-providers-google<8.4.0
+            return ClusterControllerAsyncClient(
+                credentials=self._get_credentials(), client_info=CLIENT_INFO, client_options=client_options
+            )
+        except AttributeError:
+            # for apache-airflow-providers-google>=8.4.0
+            return ClusterControllerAsyncClient(
+                credentials=self.get_credentials(),  # type: ignore[attr-defined]
```

Comment:
```suggestion
                credentials=self.get_credentials(),
```
in my opinion, to be future safe, & so that we don't need another PR to fix this mypy issue, we can have mypy fix ready for the RC once it is release and can merge this PR as soon as the RC is released

Reply:
fixed it